### PR TITLE
feat/docker bitcoind regest mode

### PR DESCRIPTION
**Pull request by vuvoth**
## Description

Closes: #XXXX

---
Add a makefile script to run and manage the bitcoind node in docker.  
Notes: This is run in regtest mode for development only.  

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
    
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

### CODE REVIEW

File: contrib/snapshot/bitcoin.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+regtest=1
+disablewallet=0
+printtoconsole=1
+rpcuser=mysecretrpcuser
+rpcpassword=mysecretrpcpassword
```
Comment (robert-zaremba) [inline contrib/snapshot/bitcoin.conf:5]:
```suggestion
rpcuser=user
rpcpassword=password
```

Comment (robert-zaremba) [reply contrib/snapshot/bitcoin.conf:5]: just to make it simpler

File: Makefile [hunk 2]
```diff
@@ -63,3 +63,44 @@ endif
...
+
+stop-bitcoind:
+	@docker stop bitcoind-node
+
+delete-bitcoind:
+	@docker rm -f bitcoind-node
+
+restart-bitcoind: delete-bitcoind create-bitcoind
+
+snapshot:
```
Comment (robert-zaremba) [inline Makefile:100]:
```suggestion
snapshot:
```
```suggestion
bitcoinregtest-snapshot:
```

File: Makefile [hunk 3]
```diff
@@ -63,3 +63,44 @@ endif
...
+        -e REGTEST=1 \
+        -e DISABLEWALLET=0 \
+        -e PRINTTOCONSOLE=1 \
+        -e RPCUSER=mysecretrpcuser \
+        -e RPCPASSWORD=mysecretrpcpassword \
+        kylemanna/bitcoind
+	@sleep 1
+	@docker exec -it bitcoind-node bitcoin-cli -regtest -rpcport=8332 loadwallet "nativewallet"
+
+create-bitcoind: snapshot run-bitcoind
```
Comment (robert-zaremba) [inline Makefile:87]:
```suggestion
bitcoind-reinit: bitcoinregtest-snapshot bitcoind-run
```

File: Makefile [hunk 4]
```diff
@@ -63,3 +63,44 @@ endif
...
+
+start-bitcoind:
```
Comment (robert-zaremba) [inline Makefile:89]:
```suggestion
bitcoind-start:
```

File: Makefile [hunk 5]
```diff
@@ -63,3 +63,44 @@ endif
...
+run-bitcoind:
+	@docker run -v $(BITCOIND_DATA):/bitcoin/.bitcoin --name=bitcoind-node -d \
+        -p 18444:8333 \
+        -p 127.0.0.1:18443:8332 \
+        -e REGTEST=1 \
+        -e DISABLEWALLET=0 \
+        -e PRINTTOCONSOLE=1 \
+        -e RPCUSER=mysecretrpcuser \
+        -e RPCPASSWORD=mysecretrpcpassword \
+        kylemanna/bitcoind
```
Comment (robert-zaremba) [inline Makefile:83]: let's define a version here to assure we all use the same version:
```
BITCOIND_CONTAINER="kylemanna/bitcoind:2400e6496045"
```

File: Makefile (hunk 4 above)
Comment (robert-zaremba) [reply Makefile:89]: in other commands, let's move bitcoind to prefix.

File: contrib/MockBitcoind.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Run bitcoind regtest node 
+
+## Create bitcoind node 
+Create a new bitcoind node with snapshot data 
+
+```bash
+make create-bitcoind
```
Comment (robert-zaremba) [inline contrib/MockBitcoind.md:8]: upate according to Makefile updates

**Review by robert-zaremba — APPROVED** [review]
pre-approving

File: contrib/MockBitcoind.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Stop bitcoind node
+
+```bash
+make stop-bitcoind
+```
+
+## Start bitcoind node
+
+You can start bitcoind node again. This command will keep any state you change on bitcoind node.
```
Comment (sczembor) [inline contrib/MockBitcoind.md:19]:
```suggestion
You can start bitcoind node again. This command will keep any state changes on the bitcoind node.
```

File: contrib/MockBitcoind.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## Restart bitcoind node
+
+Restart bitcoind will create node with snapshot data.
```
Comment (sczembor) [inline contrib/MockBitcoind.md:23]:
```suggestion
Restarting bitcoind will create node with a snapshot data.
```

**Review by sczembor — APPROVED** [review]
Looks good to me, left few suggestions

File: contrib/snapshot/bitcoin.conf (hunk 1 above)
Comment (vuvoth) [reply contrib/snapshot/bitcoin.conf:5]: Thanks! I copied this from the README of bitcoin docker 👍🏼

File: Makefile (hunk 5 above)
Comment (vuvoth) [reply Makefile]: Done!

File: contrib/MockBitcoind.md (hunk 6 above)
Comment (vuvoth) [reply contrib/MockBitcoind.md]: Done!

**Pull request merged**